tla/consensus/SIMCoverageccfraft.tla

```diff
@@ -1,65 +1,40 @@
----------- MODULE SIMCoverageccfraft ----------
-EXTENDS ccfraft, TLC, Integers, CSV, TLCExt
 
-CONSTANTS
-    NodeOne, NodeTwo, NodeThree, NodeFour, NodeFive
+  $ wget https://nightly.tlapl.us/dist/tla2tools.jar
+  $ wget https://github.com/tlaplus/CommunityModules/releases/latest/download/CommunityModules-deps.jar
+  ## Run wiht as many workers as you like to parallelize the nested simulation runs (auto uses all your cores).
+  $ java -jar tla2tools.jar -config SIMCoverageccfraft.tla SIMCoverageccfraft.tla -workers auto
 
-Servers_mc == {NodeOne, NodeTwo, NodeThree, NodeFour, NodeFive}
+----------------------------- MODULE SIMCoverageccfraft -----------------------------
+EXTENDS TLC, Naturals, Sequences, IOUtils
 
-Baseline ==
-    {<<"Next", 0..0, 0..0, 0..0>>}
+CmdLine ==
+    <<"sh", "-c",
```

Comment:
I was rather hoping we would manage to keep all the wrapping and flags in tlc.py

Reply:
The values in tlc.py do not need to be manually set because they are fine-tuned specification parameters derived from data analysis. Instead, the focus should be on creating metrics that enable TLC to automatically select the optimal probabilities. In my manual data analysis, I optimized a specific metric that measures the distance between the N-dimensional vector (composed of the distinct values for the spec's N variables) and the isotropic vector within that hyperspace (think cosine similarity, Minkowski distance, ...).  Another possible metric could be the number of distinct traces, although that metric is vulnerable to runaway variable values such as a counter.

Reply:
I don't see how there can be objective optimum here, @heidihoward will correct me if I'm wrong. There is a stable regime (replication), and transient situations (election timeouts, ack quorum losses, reconfigurations). There are two kinds of biases we can set:

1. The overall transient situations to stable regime ratio. If we go too far, the network makes no progress on average, and we're probably not testing anything interesting (that's what happened with the driver fuzzing and equally weighted actions). If we don't go far enough, we're just simulating uninteresting, happy replication, which is unlikely to spot issues.
2. The zero sum balance of weights between the transient situations. We can get real world numbers for this (how many elections take place vs reconfigurations in a typical network), but they may not be what we want to target exactly, because typical networks are probably not where issues show up.

I think some useful guidelines are probably:

1. ClientRequest should be the most frequent action, to simulate load at all times
2. An equal mix of reconfiguration and timeouts, with:
  a. Reconfiguration done in the ACL style (single node add, followed by remove, with a stable set of 3/5)
  b. Atomic reconfiguration (single node swap, stable set)
  c. Atomic whole-network reconfiguration (code update, whole set swap)

That will cover a real deployment, with elevated faults, and what we think deployments should go towards, also with elevated faults.